qa/tasks/caasp.py

```diff
@@ -0,0 +1,137 @@
+    install task.
+    One, and only one, of the machines must have a role corresponding to the
+    value of the variable salt.sm.master_role (see salt_manager.py). This node
+    is referred to as the "Salt Master", or the "master node".
+    The task starts the Salt Master daemon on the master node, and Salt Minion
+    daemons on all the nodes (including the master node), and ensures that the
+    minions are properly linked to the master. Finally, it tries to ping all
+    the minions from the Salt Master.
+    :param ctx: the argparse.Namespace object
+    :param config: the config dict
+    """
+
+    def __init__(self, ctx, config):
+        super(Caasp, self).__init__(ctx, config)
+        log.debug("beginning of constructor method")
+        self.ctx['roles'] = self.ctx.config['roles']
+        self.log = log
+        self.remotes = self.cluster.remotes
+        self.mgmt_remote = get_remote_for_role(self.ctx, "skuba_mgmt_host.0")
+        self.ssh_priv = 'tempkey.rsa'
+        self.ssh_pub = 'tempkey.rsa.pub'
+        self.set_agent = "eval `ssh-agent` && ssh-add ~/.ssh/%s && " % self.ssh_priv
+
+    def __ssh_setup(self):
+        log.debug("Executing SSH setup")
+        self.__ssh_gen_key()
+        self.__ssh_copy_priv()
+        self.__ssh_copy_pub_to_caasp()
+
+    def __ssh_gen_key(self):
+        if os.path.isfile(self.ssh_priv):
+            os.remove(self.ssh_priv)
+        if os.path.isfile(self.ssh_pub):
+            os.remove(self.ssh_pub)
+        os.system('ssh-keygen -t rsa -b 2048 -P "" -f %s' % self.ssh_priv)
+#            subprocess.check_output(['ssh-keygen', '-t', 'rsa', '-b', '2048', '-f', self.ssh_priv])
+#            res = subprocess.check_output(["ls"])
+#            log.debug(res)
+#        except subprocess.CalledProcessError as e:
+#            log.debug('error generating key')
+#            log.debug(e.output)
+
+    def __ssh_copy_priv(self):
+        log.debug("Executing __ssh_copy_priv")
+        try:
+            subprocess.check_output(['scp', self.ssh_priv, '%s:.ssh/' %
+                                     self.mgmt_remote])
+        except subprocess.CalledProcessError as e:
+            log.debug(e.output)
+
+    def __ssh_copy_pub(self, remote):
+        log.debug("Executing __ssh_copy_pub to %s" % remote.hostname)
+        cmd = "ssh-copy-id -i %s %s" % (self.ssh_pub, remote)
+        log.debug(cmd)
+        os.system(cmd)
+ #        try:
+ #            subprocess.check_output(
+ #                ['ssh-copy-id', '-i', self.ssh_pub, '%s' % remote])
+ #        except subprocess.CalledProcessError as e:
+ #            log.debug(e.output)
+
+    def __ssh_copy_pub_to_caasp(self):
+        for i in range(sum(1 for x in all_roles_of_type(
+                self.ctx.cluster, 'caasp_master'))):
+            remote = get_remote_for_role(
+                self.ctx, "caasp_master." + str(i))
+            self.__ssh_copy_pub(remote)
+        for i in range(sum(1 for x in all_roles_of_type(
```

Comment:
See comment above, there is no sense to use methods because we can, for readability it is better to use `len`.